extensions/query-store/src/settings/configureDialog.ts

```diff
@@ -0,0 +1,626 @@
+	private timeIntervalComponent?: azdata.FormComponent<azdata.Component>;
+	private aggregationSizeComponent?: azdata.FlexContainer;
+	private timeIntervalOptionsOverallReport?: string[];
+	private timeIntervalOptionsTopResourceReport?: string[];
+	private returnDataAllRadioButton?: azdata.RadioButtonComponent;
+	private returnDataTopRadioButton?: azdata.RadioButtonComponent;
+	private returnDataTopInputBox?: azdata.InputBoxComponent;
+	private returnComponent?: azdata.FormComponent<azdata.Component>;
+	private filtersInputBox?: azdata.InputBoxComponent;
+	private filterComponent?: azdata.FormComponent<azdata.Component>;
+	private initDialogComplete?: Deferred<void>;
+	private initDialogPromise: Promise<void> = new Promise<void>((resolve, reject) => this.initDialogComplete = { resolve, reject });
+
+	constructor() {
+		this.dialog = azdata.window.createModelViewDialog(constants.configure);
+		this.configureTab = azdata.window.createTab(constants.configure);
+		this.dialog.registerCloseValidator(async () => {
+			return this.validate();
+		});
+	}
+
+	async validate(): Promise<boolean> {
+		return true;
+	}
+
+	public async openDialog(): Promise<void> {
+		await this.initializeDialog();
+
+		this.dialog.okButton.label = constants.okButtonText;
+		this.dialog.okButton.enabled = false;
+		this.toDispose.push(this.dialog.okButton.onClick(async () => { }));		// TODO: Add return of settings change and functionality to enable OK Button
+
+		this.dialog.cancelButton.label = constants.cancelButtonText;
+		this.toDispose.push(this.dialog.cancelButton.onClick(async () => await this.cancel()));
+
+		let applyButton = azdata.window.createButton(constants.applyButtonText);
+		applyButton.enabled = false;
+		applyButton.onClick(async () => { });		// TODO: Add Apply click functionality and functionality to enable Apply button
+		this.dialog.customButtons = [applyButton];
+
+		azdata.window.openDialog(this.dialog);
+		await this.initDialogPromise;
+	}
+
+	protected async initializeDialog(): Promise<void> {
+		await this.initializeConfigureTab();
+		this.dialog.content = [this.configureTab];
+	}
+
+	private async initializeConfigureTab(): Promise<void> {
+		this.configureTab.registerContent(async view => {
+			this._view = view;
+
+			const consumptionCriteriaComponent = await this.createCriteriaComponent();
+			const basedOnCriteriaComponent = this.createCriteriaBasedOnComponent();
+			this.consumptionCriteriaComponent = {
+				component: this._view.modelBuilder.flexContainer()
+					.withLayout({ flexFlow: 'row' })
+					.withItems([consumptionCriteriaComponent!.component, basedOnCriteriaComponent!.component])
+					.component(),
+				title: constants.topConsumersRadioButtonsLabel
+			};
+
+			this.showChartComponent = this.createShowChartComponent();
+			this.timeIntervalComponent = this.createTimeIntervalComponent();
+			this.returnComponent = this.createReturnComponent();
+			this.filterComponent = this.createFilterComponent();
```

Comment:
do these all need to be created here or can they be created only if the dialog needs it? Maybe the report can pass in a list of the components it needs when it creates the dialog?

Reply:
Updated it to pass a list of components required and only those components are then created.